### PR TITLE
User Stories 42&43 - Merchant deactivates an item / Merchant activates an it

### PR DESCRIPTION
**Pull request by tylertomlinson**
## All Submissions:

* [x] Merge the target branch into the PR branch. Fix any conflicts that might appear.
* [x] Have you checked to ensure that other pull request are approved for merge [Pull Requests](../../../pulls)?
* [x] What stories does the PR include

### New Feature Submissions:

* [x] Have you added an explanation of what your changes do?
* [x] Have you lint your code locally prior to submission?

### Testing:

* [x] Have you written new tests for your core changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?
* [x] All new and existing tests passed.

**Test coverage** = 100% models 100% features
--------------------------------------------------------------------------------
### Change Log

Updated factory bots so associations are now correct
Testing deactivating and activating items
Added update to `Merchant::Items_Collection`
Added model test for associations on US34

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -27,7 +27,8 @@
 
   namespace :merchant do
     get "/", to: "dashboard#index"
-    get "/items", to: "items#index"
+    # get "/items", to: "items#index"
```
Comment (Mariana-21) [inline config/routes.rb]: If we're using resources then maybe we should get rid of the commented out path

File: spec/factories/merchant.rb [hunk 2]
```diff
@@ -1,6 +1,6 @@
 FactoryBot.define do
   factory :merchant, class: Merchant do
-    name     { Faker::Name.name }
+    name     { Faker::Superhero.name }
```
Comment (Mariana-21) [inline spec/factories/merchant.rb:3]: Nice!!

File: spec/factories/regular_user.rb [hunk 3]
```diff
@@ -1,6 +1,6 @@
 FactoryBot.define do
   factory :regular_user, class: User do
-    name      { Faker::Name.name }
+    name      { Faker::TvShows::BreakingBad.character }
```
Comment (Mariana-21) [inline spec/factories/regular_user.rb:3]: Sick!

**Review by Mariana-21 — APPROVED** [review]
Really great testing!

File: app/views/items/index.html.erb [hunk 4]
```diff
@@ -30,15 +30,15 @@
     <h2> <%=link_to item.name, "/items/#{item.id}" %> </h2>
     <p>Sold by: <%=link_to item.merchant.name, "/merchants/#{item.merchant.id}" %></p>
     <%= link_to image_tag(item.image), "/items/#{item.id}", id: "img-#{item.id}" %>
-    <p> <%= item.description unless @merchant%> </p>
+    <p> <%= item.description if current_merchant? || !@merchant %> </p>
     <p>Price: <%=number_to_currency(item.price) %> </p>
     <p>Inventory: <%= item.inventory %> </p>
-    <% if !@merchant %>
-    <% end %>
```
Comment (alex-latham) [inline app/views/items/index.html.erb:37]: Redundant!

File: spec/factories/item.rb [hunk 5]
```diff
@@ -1,6 +1,6 @@
 FactoryBot.define do
   factory :item, class: Item do
-    name        { Faker::Name.name}
+    name        { Faker::Commerce.product_name}
```
Comment (alex-latham) [inline spec/factories/item.rb:3]: Ok, this makes things a bit clearer!

File: spec/factories/item_order.rb [hunk 6]
```diff
@@ -2,5 +2,7 @@
   factory :item_order, class: ItemOrder do
     price        { Faker::Commerce.price }
     quantity     { Faker::Number.between(from: 1, to: 200) }
+    association  :item, factory: :item
+    association  :order, factory: :order
```
Comment (alex-latham) [inline spec/factories/item_order.rb:6]: !!!

**Review by alex-latham — APPROVED** [review]
These changes looks good

**Pull request merged**